ci/install.sh

```diff
@@ -99,35 +102,127 @@ if [ ! -e "${GPG21_INSTALL}/bin/gpg" ]; then
+  cd pinentry-${PINENTRY_VERSION}
+  ./configure --prefix="${GPG_INSTALL}" \
+    --with-libgpg-error-prefix="${GPG_INSTALL}" \
+    --with-libassuan-prefix="${GPG_INSTALL}" \
     --enable-pinentry-curses \
     --disable-pinentry-qt4
   ${MAKE} -j${CORES} install
   cd ..
 
-  wget -c https://www.gnupg.org/ftp/gcrypt/gnupg/gnupg-2.2.4.tar.bz2
-  wget -c https://www.gnupg.org/ftp/gcrypt/gnupg/gnupg-2.2.4.tar.bz2.sig
-  gpg --verify gnupg-2.2.4.tar.bz2.sig
-  tar -xjf gnupg-2.2.4.tar.bz2
-  cd gnupg-2.2.4
-  ./configure --prefix="${GPG21_INSTALL}" \
-    --with-libgpg-error-prefix="${GPG21_INSTALL}" \
-    --with-libgcrypt-prefix="${GPG21_INSTALL}" \
-    --with-libassuan-prefix="${GPG21_INSTALL}" \
-    --with-ksba-prefix="${GPG21_INSTALL}" \
-    --with-npth-prefix="${GPG21_INSTALL}"
+  wget -c https://www.gnupg.org/ftp/gcrypt/gnupg/gnupg-${GNUPG_VERSION}.tar.bz2
+  wget -c https://www.gnupg.org/ftp/gcrypt/gnupg/gnupg-${GNUPG_VERSION}.tar.bz2.sig
+  gpg --verify gnupg-${GNUPG_VERSION}.tar.bz2.sig
+  tar -xjf gnupg-${GNUPG_VERSION}.tar.bz2
+  cd gnupg-${GNUPG_VERSION}
+  ./configure --prefix="${GPG_INSTALL}" \
+    --with-libgpg-error-prefix="${GPG_INSTALL}" \
+    --with-libgcrypt-prefix="${GPG_INSTALL}" \
+    --with-libassuan-prefix="${GPG_INSTALL}" \
+    --with-ksba-prefix="${GPG_INSTALL}" \
+    --with-npth-prefix="${GPG_INSTALL}"
   ${MAKE} -j${CORES} install
 
+}
+
+build_gpg_beta() {
+  mkdir gettext && cd gettext
+  gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys D7E69871
+  wget -c https://ftp.gnu.org/pub/gnu/gettext/gettext-latest.tar.xz
+  wget -c https://ftp.gnu.org/pub/gnu/gettext/gettext-latest.tar.xz.sig
+  gpg --verify gettext-latest.tar.xz.sig
+  tar -xJf gettext-latest.tar.xz --strip 1
+  ./configure --prefix="${GPG_INSTALL}"
+  ${MAKE} -j${CORES} install
+  cd ..
+  export GETTEXT_PREFIX="${GPG_INSTALL}/bin/"
+
+  # workaround https://github.com/travis-ci/travis-ci/issues/8613
+  # alternatively, forcing the libgpg-error build to use gcc should work
+  export LD_LIBRARY_PATH="/usr/local/clang-5.0.0/lib"
+  for repo in npth libgpg-error; do
+    git clone git://git.gnupg.org/${repo}
+    cd ${repo}
+    ./autogen.sh
+    ./configure --prefix="${GPG_INSTALL}"
+    ${MAKE} -j${CORES} install
+    cd ..
+  done
+
+  for repo in libgcrypt libassuan libksba; do
+    git clone git://git.gnupg.org/${repo}
+    cd ${repo}
+    ./autogen.sh
+    ./configure --prefix="${GPG_INSTALL}" --disable-doc --with-libgpg-error-prefix="${GPG_INSTALL}"
+    ${MAKE} -j${CORES} install
+    cd ..
+  done
+
+  git clone git://git.gnupg.org/pinentry.git
+  cd pinentry
+  cat << 'END' | git apply -
+diff --git a/Makefile.am b/Makefile.am
+index 8c8b8e5..412244c 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -85,7 +85,7 @@ endif
+ SUBDIRS = m4 secmem pinentry ${pinentry_curses} ${pinentry_tty} \
+ 	${pinentry_emacs} ${pinentry_gtk_2} ${pinentry_gnome_3} \
+ 	${pinentry_qt} ${pinentry_tqt} ${pinentry_w32} \
+-	${pinentry_fltk} doc
```

Comment:
Just a note: I realize this is not ideal and will likely require modification at some point, but I couldn't see an easy alternative. pinentry doesn't have a `--disable-doc` configure option like some of the others do, and it requires `makeinfo`, and the version of `makeinfo` in the ubuntu:trusty `texinfo` package seems to be too old.